engine/runtime/src/main/java/org/enso/interpreter/runtime/EnsoContext.java

```diff
@@ -774,6 +774,17 @@ public TruffleLogger getLogger(Class<?> klass) {
     return TruffleLogger.getLogger(LanguageInfo.ID, klass);
   }
 
+  /**
+   * Gets a logger for the specified class that is bound to this engine.
+   * Such logger may then be safely used in threads defined in a thread-pool.
+   *
+   * @param clazz the class to name log entries with
+   * @return a new logger for the specified {@code path}
+   */
+  public TruffleLogger getLoggerBoundToEngine(Class<?> clazz) {
+    return environment.getLogger(clazz);
```

Comment:
Let's just use this in the existing `getLogger` method without introducing a new one.

Reply:
I guess this is the core of the fix. It seems correct.